amethyst_ui/src/selection.rs

```diff
@@ -0,0 +1,268 @@
+use amethyst_core::{
+    shrev::EventChannel,
+    specs::{
+        Component, DenseVecStorage, Read, ReadStorage, ReaderId, Resources, System, SystemData,
+        WriteStorage,
+    },
+};
+use amethyst_input::InputHandler;
+use amethyst_renderer::{ElementState, Event, KeyboardInput, VirtualKeyCode, WindowEvent};
+use std::{hash::Hash, marker::PhantomData};
+
+use {CachedSelectionOrder, UiEvent, UiEventType};
+
+/// Component indicating that a Ui entity is selectable.
+/// Generic Type:
+/// - G: Selection Group. Used to determine which entities can be selected together at the same time.
+#[derive(Debug, Serialize, Deserialize, new)]
+pub struct Selectable<G> {
+    /// The order in which entities are selected when pressing the `Tab` key or the "go to next" input action.
+    pub order: u32,
+    #[new(default)]
+    /// A multi selection group. When multiple entities are in the same selection group, they can be selected at
+    /// the same time by holding shift or control and clicking them.
+    /// You can also select the first element, then hold shift and press the keyboard arrow keys.
+    // TODO: Holding shift + arrow keys to select more.
+    // TODO: Pressing the arrow keys could optionally be binded to change the selected ui element.
+    pub multi_select_group: Option<G>,
+    #[new(default)]
+    /// Indicates if you can select multiple entities at once without having to press the shift or control key.
+    pub auto_multi_select: bool,
+    /// Indicates if this requires the inputs (except Tab) be ignored when the component is focused.
+    #[new(default)]
+    pub require_input: bool,
```

Comment:
Am I understanding correctly that with this option will basically "swallow" the inputs when the component is focused? If so I find the name a little confusing, shouldn't it rather be `swallows_inputs` or `consumes_inputs` or `stops_propagation` or something?